### PR TITLE
Small change to return error from periodic exporter

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

The `collect_and_export()` drops the error in case of failure during export, and always return success. Fixing that. Testing using end-to-end test for OTLP metrics exporter, with no collector running. Not adding the unit-test for now - can be done by creating a failing MockExporter (if required).

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/periodic_reader.rs [hunk 1]
```diff
@@ -260,7 +260,7 @@ impl<RT: Runtime> PeriodicReaderWorker<RT> {
         pin_mut!(timeout);
 
         match future::select(export, timeout).await {
-            Either::Left(_) => Ok(()),
+            Either::Left((res, _)) => res, // return the result.
```
Comment (cijothomas) [inline opentelemetry-sdk/src/metrics/periodic_reader.rs:263]: good catch! might be worth adding as a bug fix chanelog entry.

**Review by cijothomas — APPROVED** [review]

**Review by TommyCpp — APPROVED** [review]
Good catch

File: opentelemetry-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -45,6 +45,8 @@
 
 - **Breaking** [1850] (https://github.com/open-telemetry/opentelemetry-rust/pull/1850) `LoggerProvider::log_processors()` and `LoggerProvider::resource()` are not public methods anymore. They are only used within the `opentelemetry-sdk` crate.
 
+- [1857](https://github.com/open-telemetry/opentelemetry-rust/pull/1857) Fixed an issue in the `collect_and_export()` method of the periodic exporter where errors during export were dropped, ensuring proper error handling and reporting.
```
Comment (cijothomas) [inline opentelemetry-sdk/CHANGELOG.md:48]: `collect_and_export()` is not a public method. I'd suggest to phrase this in a way applicable to end users.

something like:
```Fixed an issue in Metrics SDK which prevented export errors from being send to global error handler. With the fix, errors occurring during export like OTLP Endpoint unresponsive shows up in stderr by default.```

Comment (lalitb) [reply opentelemetry-sdk/CHANGELOG.md]: Yes true. Updated.

**Pull request merged**